src/main/java/com/google/api/codegen/tools/DiscoToolDriverBase.java

```diff
@@ -0,0 +1,76 @@
+/* Copyright 2017 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.tools;
+
+import com.google.api.codegen.DiscoGapicGeneratorApi;
+import com.google.api.codegen.DocumentGenerator;
+import com.google.api.codegen.discovery.Document;
+import com.google.api.tools.framework.model.Diag;
+import com.google.api.tools.framework.model.SimpleLocation;
+import com.google.api.tools.framework.tools.GenericToolDriverBase;
+import com.google.api.tools.framework.tools.ToolOptions;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/**
+ * Abstract base class for drivers running tools based on the framework.
+ *
+ * <p>Uses {@link ToolOptions} to pass arguments to the driver.
+ */
+public abstract class DiscoToolDriverBase extends GenericToolDriverBase {
```

Comment:
I'd prefer not to have this functionality as an extra layer in the inheritance. Could we push the functionality into a utility class?

Reply:
Done.